src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/ADAMInstance.cs

```diff
@@ -622,7 +622,7 @@ public string DefaultPartition
                         }
                         else
                         {
-                            _cachedDefaultPartition = (string)PropertyManager.GetPropertyValue(context, ntdsaEntry, PropertyManager.MsDSDefaultNamingContext);
+                            _cachedDefaultPartition = (string)PropertyManager.GetPropertyValue(context, ntdsaEntry, PropertyManager.MsDSDefaultNamingContext)!;
```

Comment:
Why is this `!` needed?  Should it instead be casting to `(string?)`?

Reply:
it doesn't really matter in this case but because there is null check above it shouldn't be null.